mod_reforged/hooks/misc.nut

```diff
@@ -4,3 +4,65 @@
 	if (_function != null) _function(obj);
 	return obj;
 }
+
```

Comment:
I would prefer this to be a static map of {WeaponType : pg.id} and then a loop of the weaponTypes of the weapon.

Reply:
```
::Reforged.WeaponTypeToPerkGroupsMap <- {
	::Const.Items.WeaponType.Axe = "pg.rf_axe",
	::Const.Items.WeaponType.Bow = "pg.rf_bow",
	::Const.Items.WeaponType.Cleaver = "pg.rf_cleaver",
	::Const.Items.WeaponType.Dagger = "pg.rf_dagger",
	::Const.Items.WeaponType.Flail = "pg.rf_flail",
	::Const.Items.WeaponType.Hammer = "pg.rf_hammer",
	::Const.Items.WeaponType.Mace = "pg.rf_mace",
	::Const.Items.WeaponType.Polearm = "pg.rf_polearm",
	::Const.Items.WeaponType.Spear = "pg.rf_spear",
	::Const.Items.WeaponType.Sword = "pg.rf_sword",
	::Const.Items.WeaponType.Throwing = "pg.rf_throwing",
	::Const.Items.WeaponType.Crossbow ="pg.rf_crossbow",
	::Const.Items.WeaponType.Firearm ="pg.rf_crossbow",
} 

::Reforged.getPerkGroupsOfWeapon <- function( _weapon )
{
	local ids = [];

	if (_weapon == null || !_weapon.isItemType(::Const.Items.ItemType.Weapon))
	{
		return ids;
	}
	foreach(weaponType in _weapon.getWeaponTypes())
	{
		ids.push(weaponType);
	}
	return ids;
}

```
I don't know if getWeaponTypes exists.

Reply:
Good suggestion, I included them in the new commits